hermes/src/db/local_cache.rs

```diff
@@ -0,0 +1,88 @@
+use {
+    super::{
+        Db,
+        DbRecord,
+        RequestTime,
+    },
+    anyhow::Result,
+    dashmap::DashMap,
+    std::{
+        collections::VecDeque,
+        sync::Arc,
+    },
+};
+
+#[derive(Clone)]
+pub struct LocalCache {
+    cache:            Arc<DashMap<Vec<u8>, VecDeque<DbRecord>>>,
+    max_size_per_key: usize,
+}
+
+impl LocalCache {
+    pub fn new(max_size_per_key: usize) -> Self {
+        Self {
+            cache: Arc::new(DashMap::new()),
+            max_size_per_key,
+        }
+    }
+}
+
+impl Db for LocalCache {
+    /// Add a new db record to the deque for the key while keeping it sorted.
+    /// This function also removes the oldest record in the cache if the max_size is reached.
+    /// Records are usually added in increasing order and the function is optimized for that case.
```

Comment:
```suggestion
    /// Add a new db entry to the cache.
    /// 
    /// This method keeps the backed store sorted for efficiency, and removes
    /// the oldest record in the cache if the max_size is reached. Entries are
    /// usually added in increasing order and likely to be inserted near the
    /// end of the deque. The function is optimized for this specific case.
```

Added a single space between the one-line description and the body, we should maintain this style for all doc-style comments because it more closely reflects what Rustdoc looks like.